Sofa/framework/Type/test/VecTypes_test.cpp

```diff
@@ -23,6 +23,40 @@
 #include <sofa/defaulttype/VecTypes.h>
 #include <gtest/gtest.h>
 
+template < sofa::Size N, typename ValueType>
+void defaultConstructor()
+{
+    const sofa::type::Vec<N, ValueType> vec;
+    for (const auto& value : vec)
+    {
+        EXPECT_EQ(value, ValueType());
```

Comment:
Is this creator defined even for non-object types like float and int ?   
And are we sure this throw zeros ? 

Using simple cast might do the job and be more explicit ?

Reply:
This is the most generic way to instantiate a type with its default constructor. In case of primitive types, it initializes the value to zero.
Since I test only primitive types, a static_cast on zero can be used instead. I don't mind changing it if the reviewers prefer.